CMakeModules/Dependencies.cmake

```diff
@@ -51,7 +51,12 @@ macro(resolve_dependencies) # Intentionally a macro - we want set() to work in p
             set(PREBUILT_DEPS_BUILD_TYPE "Release")
         endif()
 
-        set(PREBUILT_DEPS_FILENAME "${BUILD_PLATFORM}_${PREBUILT_DEPS_BUILD_TYPE}_${BUILD_ARCHITECTURE}.zip")
+        #OSX cross compile support 
+        if(APPLE)
+            set(PREBUILT_DEPS_FILENAME "${BUILD_PLATFORM}_${PREBUILT_DEPS_BUILD_TYPE}_${CMAKE_OSX_ARCHITECTURES}.zip")
+        else()
+            set(PREBUILT_DEPS_FILENAME "${BUILD_PLATFORM}_${PREBUILT_DEPS_BUILD_TYPE}_${BUILD_ARCHITECTURE}.zip")
+        endif()
```

Comment:
This looks hacky to me. Can we fix this by properly setting `BUILD_ARCHITECTURE` instead?

Reply:
I dont disagree its a bit wierd how osx does things, problem is osx allows things like, CMAKE_OSX_ARCHITECTURES:STRING="x86_64;arm64" for universal binaries, so it appears that BUILD_ARCHITECTURE  isnt really used on osx with cmake

We could have a intermediary value that we set and if its set we use it for the dl string, else use BUILD_ARCHITECTURE that doesnt really seem much better tho... thoughts?

Reply:
After some looking seems that BUILD_ARCHITECTURE isnt even documented... looks like its some under the hood plumbing i guess?